src/models/goods-vehicle-movements.js

```diff
@@ -41,13 +38,38 @@ const mapCustomsDeclarations = (
 ) => {
   const gmrCustoms = goodsVehicleMovement?.declarations?.customs?.map((custom) => {
     return mapGmrDeclaration(customsDeclarations, custom)
-  })
+  }) || []
 
   const gmrTransits = goodsVehicleMovement?.declarations?.transits?.map((transit) => {
     return mapGmrDeclaration(customsDeclarations, transit)
-  })
+  }) || []
+
+  emitMetrics(gmrCustoms, gmrTransits)
+
+  return (gmrCustoms).concat(gmrTransits)
+}
+
+const mrnCounter = (counter, custom, shouldBeCounted) => {
+  if (custom.isKnownMrn === shouldBeCounted) {
+    ++counter
+  }
 
-  return (gmrCustoms || []).concat(gmrTransits || [])
+  return counter
+}
+
+const emitMetrics = (gmrCustoms, gmrTransits) => {
+  const knownMrns = gmrCustoms.reduce((knownMrnsCount, custom) => mrnCounter(knownMrnsCount, custom, true), 0)
+    + gmrTransits.reduce((knownMrnsCount, custom) => mrnCounter(knownMrnsCount, custom, true), 0)
+
+  const unknownCustomsMrns = gmrCustoms.reduce((unknownMrnsCount, custom) => mrnCounter(unknownMrnsCount, custom, false), 0)
+
+  if (knownMrns > 0) {
+    metricsCounter(metricName.GMR_KNOWN_MRNS, knownMrns)
```

Comment:
let's move all invocations of the `metricsCounter` function to `gmr-search-result.js`
The `models` code is intended to only project objects from one form to another.

This function can be renamed and still provide information about known and unknown MRNs
e.g.
`getGvmDeclarationsKnownStatus()` and return something along the lines of `{ knownMrns: 2, unknownMrns: 1 }`